### ISSUE TITLE
Mapping for depends_on and ContainerDependency

### ISSUE THREAD

**Issue by dkovvuri** [aws/amazon-ecs-cli#897]
### Support transformation of [depends_on](https://docs.docker.com/compose/compose-file/compose-file-v2/#depends_on) in Docker Compose to [ContainerDependency](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#container_definition_dependson) objects in the ECS task definition.

ECS has recently announced support for [container dependency](https://aws.amazon.com/about-aws/whats-new/2019/03/amazon-ecs-introduces-enhanced-container-dependency-management/) within a task. When using the ecs-cli to transform a docker-compose file to an ECS task definition the options ***depends_on*** are skipped with the error:

>WARN[0000] Skipping unsupported YAML option for service...  option name=depends_on service name=webserver

#### Steps to Replicate

I. Docker-Compose File

```
version: '2'
services:
    redis:
        image: 'redis:3.2.7'

    postgres:
        image: postgres:9.6
        environment:
            - POSTGRES_USER=airflow
            - POSTGRES_PASSWORD=airflow
            - POSTGRES_DB=airflow

    webserver:
        image: puckel/docker-airflow:1.10.3
        restart: always
        links:
            - postgres
            - redis
        depends_on:
            - postgres
            - redis
        ports:
            - "8080:8080"
        command: webserver
```

II. Run `ecs-cli`

```
ecs-cli compose --file compose.yml --project-name test run --region us-east-1
```

**Comment by PettitWesley** [comment]
I like this idea; depends_on could be mapped to a container dependency with condition `START`.

**Comment by robbymarston** [comment]
+1, I would find this incredibly useful!

**Comment by PavanYadala** [comment]
How this issue can be dealt with?

**Comment by PavanYadala** [comment]
> ### Support transformation of [depends_on](https://docs.docker.com/compose/compose-file/compose-file-v2/#depends_on) in Docker Compose to [ContainerDependency](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#container_definition_dependson) objects in the ECS task definition.
> ECS has recently announced support for [container dependency](https://aws.amazon.com/about-aws/whats-new/2019/03/amazon-ecs-introduces-enhanced-container-dependency-management/) within a task. When using the ecs-cli to transform a docker-compose file to an ECS task definition the options _**depends_on**_ are skipped with the error:
> 
> > WARN[0000] Skipping unsupported YAML option for service...  option name=depends_on service name=webserver
> 
> #### Steps to Replicate
> I. Docker-Compose File
> 
> ```
> version: '2'
> services:
>     redis:
>         image: 'redis:3.2.7'
> 
>     postgres:
>         image: postgres:9.6
>         environment:
>             - POSTGRES_USER=airflow
>             - POSTGRES_PASSWORD=airflow
>             - POSTGRES_DB=airflow
> 
>     webserver:
>         image: puckel/docker-airflow:1.10.3
>         restart: always
>         links:
>             - postgres
>             - redis
>         depends_on:
>             - postgres
>             - redis
>         ports:
>             - "8080:8080"
>         command: webserver
> ```
> 
> II. Run `ecs-cli`
> 
> ```
> ecs-cli compose --file compose.yml --project-name test run --region us-east-1
> ```
Where should i mention the depends_on option. In doker-compose.yml or ecs-parms.yml? Also please let me know the format as well. I want my web app should be started before my nginx server starts.

**Comment by Sylvestre67** [comment]
Is there any way to set of the `dependsOn` array of a `task_definition.container_definitions` using `ecs-cli` at this moment ?

**Comment by rohitg1594** [comment]
Any updates on this? I would really love to have this feature.

**Comment by ekanai** [comment]
When this issue will be supported?

**Comment by diablodale** [comment]
I think customers will have to do this without the ecs-cli team. They wrote yesterday, "We’ve shifted our resources over the past few months to developing [ecs-cli] v2..."
https://github.com/aws/containers-roadmap/issues/513#issuecomment-567280886

With that in mind, I suggest someone take on this code fix themselves. Fork the repo and submit a PR. We can share the PR in our private forks of this repo until/if ecs-cli team integrates the PR.

**Comment by allisaurus** [comment]
Closing, as this feature is now available in [v1.21.0](https://github.com/aws/amazon-ecs-cli/releases/tag/v1.21.0) !
